### PR TITLE
feat:Creation of new doc on security

### LINKED ISSUE #13305: Docs: Create new doc on security
## Summary

Security in Gatsby is an important topic, and deserves a page in the docs describing what Gatsby does well and what developers need to watch out for. It could go under "Improving Performance" in the Guides section of the docs.

There is a wonderful new blog post on security from @moonmeister: https://www.gatsbyjs.org/blog/2019-04-06-security-for-modern-web-frameworks/

This issue serves as a follow-up to make sure we add a dedicated page in the docs for security in Gatsby. It should list best practices and gotchas to prepare Gatsby devs as much as possible for security limitations and requirements.

### PR DESCRIPTION
**Pull request by kushthedude**
## Description

- Adding documentation about security

## Related Issues

Fixes #13305

### CODE REVIEW

**Comment by kushthedude** [comment]
@marcysutton @sidharthachatterjee @KyleAMathews Please review and suggest changes to the following.

**Comment by kushthedude** [comment]
@pieh Please Review

**Comment by kushthedude** [comment]
@sidharthachatterjee Please check

**Comment by LekoArts** [comment]
@kushthedude Please stop mentioning/pinging people after 5 hours of opening your PR. We're considering this rude behavior (please read our Code of Conduct: https://www.gatsbyjs.org/contributing/code-of-conduct/).

We're trying to look at all PRs in a timely manner but your PR is not the only one that needs to be looked at. Furthermore, you're probably in another timezone so please also think about that. Thanks!

**Comment by kushthedude** [comment]
> @kushthedude Please stop mentioning/pinging people after 5 hours of opening your PR. We're considering this rude behavior (please read our Code of Conduct: https://www.gatsbyjs.org/contributing/code-of-conduct/).
> 
> We're trying to look at all PRs in a timely manner but your PR is not the only one that needs to be looked at. Furthermore, you're probably in another timezone so please also think about that. Thanks!



> @kushthedude Please stop mentioning/pinging people after 5 hours of opening your PR. We're considering this rude behavior (please read our Code of Conduct: https://www.gatsbyjs.org/contributing/code-of-conduct/).
> 
> We're trying to look at all PRs in a timely manner but your PR is not the only one that needs to be looked at. Furthermore, you're probably in another timezone so please also think about that. Thanks!

The number of PRs made me realise the way of reviewing, ANYWAYS you guys are the repository maintainers here. So please go ahead according the COC.

**Comment by kushthedude** [comment]
Also I have already read the COC for Gatsby before contributing to it,
hence I dont think it is anywhere wrong to ask for review by pinging.

File: docs/docs/security-in-gatsby.md [hunk 1]
```diff
@@ -1,9 +1,137 @@
...
 ---
 
-This is a stub. Help our community expand it.
+Among the many lauded benefits of using Gatsby (and other static app frameworks) is security. It is encouraging to see a framework not require developers to stress about security, but for those new to Gatsby or web development, this can contribute to a myth that there are no security issues.
 
-Please use the [Gatsby Style Guide](/contributing/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+There are indeed whole classes of security vulnerabilities that are eliminated by using Gatsby for web development; but there are still other vulnerabilities that are shared and some that are created. As a Gatsby developer, you need to make sure you are aware of these possible vulnerabilities and how to remedy them.
+
+This post will act as an introduction to get you familiar with these concepts in Gatsby.
```
Comment (laurieontech) [inline docs/docs/security-in-gatsby.md]:
```suggestion
This post will act as an introduction to familiarize yourself with these concepts in Gatsby.
```

File: docs/docs/security-in-gatsby.md [hunk 2]
```diff
@@ -1,9 +1,137 @@
...
+
+## Defining Terms
+
+These are not to be considered universal definitions, just how the author will use them going forward.
+
+- **site** - Any website, web-app, or other content you can browse on the internet in a web browser.
+
+- **server-side site** - Any site that pre-processes code at time of request -- whether using node.js, PHP, Python, or any other method -- before delivering it to the user's browser.
+
+- **client-side site** - Any site (GatsbyJS, NuxtJS, NextJS, Single Page App, etc...) where static files(HTML/CSS/JS) are shipped to the browser for processing -- pre-processing may happen at build time, but does not happen time of request.
```
Comment (laurieontech) [inline docs/docs/security-in-gatsby.md]:
```suggestion
- **client-side site** - Any site (GatsbyJS, NuxtJS, NextJS, Single Page App, etc...) where static files(HTML/CSS/JS) are shipped to the browser for processing -- pre-processing may happen at build time, but does not happen at the time of request.
```

**Comment by kushthedude** [comment]
@laurieontech commited the changes

**Comment by laurieontech** [comment]
Hey @kushthedude, thanks for the PR! Lots of good info here. With such a large document it's going to take us a bit to complete the review, so hang tight.

**Comment by kushthedude** [comment]
Sure hanging in there for the review👍

**Comment by moonmeister** [comment]
This is just copied from the the [blog post](https://www.gatsbyjs.org/blog/2019-04-06-security-for-modern-web-frameworks/) I wrote on the Gatsby blog and is fairly blatant plagiarism. That blog post is why issue  #13305 exists (because we realized there was a lack of security content surrounding Gatsby). No one at the time of my writing that post expected my post to become that security document and there is no benefit to wholesale copying content from one URL to another. 

If the Gatsby team feels it'd be beneficial to use the content of my post as the basis of the security document that's fine, but I'd still expect at least some basic re-organization and modification to make it read like documentation and not a blog post.

**Comment by kushthedude** [comment]
I did make some changes to remove the Gist of blog post from the following.
It would be great of there some suggestion you can give

**Comment by moonmeister** [comment]
The only two commits I see from your original commit are you committing the suggestions made by @laurieontech. You could start by showing your work. It'll take some finagling, but including, in a commit, all the changes you made from the original post would be acceptable to show your work for now.

That said, based on my cherry picking various sentences and reviewing the general outline, I did not see any changes (aka. plagiarism). 

What actually needs to happen is you need to start from the beginning, writing your own content or at least organizing and collating content . If you want to quote me directly do so, if you want to pull ideas from my work or others that is fine. But just like I did in my post, you need to use footnotes or inline references to **give credit to the original authors**. 

@marcysutton gave a sentence or two description of what she expected in the original issue #13305. I suggest re-reading that and starting from there. 

Ultimately, you need to be knowledgeable enough on security to be able to put together this security content, it is okay you are not, but plagiarism of my content or anyone else's content is not acceptable.

**Comment by kushthedude** [comment]
👍 Dont worry, I will remove all of your content. Thanks for the concern.

**Comment by m-allanson** [comment]
Hey @kushthedude - please don't plagiarise the work of others. That is unacceptable behaviour here. 

The content of this PR is almost identical to @moonmeister's blog post. As your PR is a) duplicate content and b) does not credit the original author I am closing it. 

We'd love to see a PR that creates this doc, but it _must_ be original content and credit any references where needed.

**Pull request closed without merging**